src/vs/platform/remoteTunnel/electron-browser/remoteTunnelService.ts

```diff
@@ -285,8 +286,19 @@ export class RemoteTunnelService extends Disposable implements IRemoteTunnelServ
 
 	private _getHostName(): string | undefined {
 		let name = this.configurationService.getValue<string>(CONFIGURATION_KEY_HOST_NAME) || hostname();
-		name = name.replace(/[^\w-]/g, '').substring(0, 20);
+		name = _removeLeadingDashes(name).replace(/[^\w-]/g, '').substring(0, 20);
```

Comment:
I believe `replace(/^-+/g, '')` is shorter and faster

Reply:
Ah, nice, I didn't think of that. New PR on the way.

Reply:
https://github.com/microsoft/vscode/pull/168975